kubernetes/structure_job.go

```diff
@@ -112,6 +123,152 @@ func expandJobV1Spec(j []interface{}) (batchv1.JobSpec, error) {
 	return obj, nil
 }
 
+func expandPodFailurePolicy(l []interface{}) *batchv1.PodFailurePolicy {
+	obj := &batchv1.PodFailurePolicy{}
+	if len(l) == 0 || l[0] == nil {
+		return obj
+	}
+	in := l[0].(map[string]interface{})
+
+	if v, ok := in["rule"].([]interface{}); ok && len(v) > 0 {
+		rules := expandPodFailurePolicyRules(v)
+		obj.Rules = rules
+	}
+	return obj
+}
+
+func expandPodFailurePolicyRules(l []interface{}) []batchv1.PodFailurePolicyRule {
+	obj := make([]batchv1.PodFailurePolicyRule, len(l))
+	if len(l) == 0 || l[0] == nil {
+		return obj
+	}
+	for i, rule := range l {
+		objRule := &batchv1.PodFailurePolicyRule{}
+
+		r := rule.(map[string]interface{})
+
+		if v, ok := r["action"].(string); ok && v != "" {
+			objRule.Action = batchv1.PodFailurePolicyAction(v)
+		}
+
+		if v, ok := r["on_exit_codes"].([]interface{}); ok && len(v) > 0 {
+			onExitCodes := expandPodFailurePolicyOnExitCodesRequirement(v)
+			objRule.OnExitCodes = onExitCodes
+		}
+
+		if v, ok := r["on_pod_condition"].([]interface{}); ok && len(v) > 0 {
+			podConditions := expandPodFailurePolicyOnPodConditionsPattern(v)
+			objRule.OnPodConditions = podConditions
+		}
+
+		obj[i] = *objRule
+	}
+
+	return obj
+}
+
+func expandPodFailurePolicyOnExitCodesRequirement(l []interface{}) *batchv1.PodFailurePolicyOnExitCodesRequirement {
+	obj := &batchv1.PodFailurePolicyOnExitCodesRequirement{}
+	if len(l) == 0 || l[0] == nil {
+		return obj
+	}
+	in := l[0].(map[string]interface{})
+
+	if v, ok := in["container_name"].(string); ok && v != "" {
+		obj.ContainerName = &v
+	}
+
+	if v, ok := in["operator"].(string); ok && v != "" {
+		obj.Operator = batchv1.PodFailurePolicyOnExitCodesOperator(v)
+	}
+
+	if v, ok := in["values"].(*schema.Set); ok && v != nil {
```

Comment:
Because you've changed the type to `TypeList` you'll need to change this logic too.

Reply:
running into a blocker where setting `"values"` to `TypeList` is resulting in `in["values"]` being empty. this can be seen even before it gets to the `"values"` check.

```hcl
│ Error: Failed to create Job! API error: Job.batch "policy-test" is invalid: spec.podFailurePolicy.rules[0].onExitCodes.values: Invalid value: []int32(nil): at least one value is required
```

So it passes the schema checks of `MinItems` but somewhere it's losing the actual values from the list.